fetchmedia.go

```diff
@@ -14,9 +14,35 @@ import (
 	"github.com/vincent-petithory/dataurl"
 )
 
+// grabber is a top level structure that allows a custom http client.
+// if onlyChecl is true, the methods will not perform actual grab to spare memory and bandwidth
+type grabber struct {
```

Comment:
I'm wondering if we really need a separate `grabber` struct if all it does is hold a reference to the http client? I think part of the reason I'm not sure about this is that the name "grabber" isn't very meaningful, so it feels like it adds confusion but it doesn't really seem to save any code from what I can tell, other than passing the HTTP client around. but that's only used for the http handler; the other handlers don't even need it, right?

Reply:
In my first implementation grabber had other elements.
In Go, a structure+method is syntactic sugar that replaces a function argument (internally the behavior is the same).
I prefer this option because in the constructor we can set the default client as we wish; otherwise we move the decision to the function calling the `fetch` methods.

Besides that I do not have any opinion about the name.

Reply:
I was actually thinking why not just create the HTTP client inside `httpHandler` since that's the only place in the code it's used. However I just read this:

> Clients and Transports are safe for concurrent use by multiple goroutines and for efficiency should only be created once and re-used. 

https://pkg.go.dev/net/http#pkg-overview

So it does seem that having one HTTP client for the whole library is more efficient, and as you mentioned this is just syntactic sugar for passing it around without having to have yet another function argument. LGTM.